File: arcade/matrixElementsSum.js
function matrixElementsSum(matrix) {
    var sum = 0;
    for (var i = 0; i <matrix.length;i++){
        for (var j = 0; j <matrix[i].length;j++){
            if (matrix[i][j] == 0)
                continue;
            else if ((i>0) && (i-1 >= 0) && (matrix[i-1][j] == 0))
                matrix[i][j] = 0;
            else
                sum = sum + matrix[i][j];
        }
    }
    return sum;
}

File: arcade/matrixElementsSum.py
def matrixElementsSum(matrix):
    sum = 0
    for i in range (len(matrix)):
        for j in range (len(matrix[i])):
            if (matrix[i][j] == 0):
                pass
            elif (i>0) and (i-1 >= 0) and (matrix[i-1][j] == 0):
                matrix[i][j] = 0
            else:
                sum = sum + matrix[i][j]
                print 'result',matrix[i][j]
    return sum
